### PR TITLE
Stub out ResolveComReference task to give proper error message during…

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
… dotnet-sdk build.

See #3986

cc @rainersigwald

### CODE REVIEW

File: src/Tasks/Resources/Strings.resx [hunk 1]
```diff
@@ -2673,6 +2673,12 @@
     <comment>{StrBegin="MSB4801: "}</comment>
   </data>
 
+  <!-- Some tasks are only supported on .NET Framework -->
+  <data name="TaskRequiresFrameworkFailure" xml:space="preserve">
+    <value>MSB4803: The task "{0}" is not supported on the .NET Core version of MSBuild. Please use the .NET Framework version of MSBuild.</value>
```
Comment (rainersigwald) [inline src/Tasks/Resources/Strings.resx]: @kathleendollard @livarcocc  Thoughts on wording here? I don't know of a good customer-facing way to distinguish between .NET Core MSBuild and full framework MSBuild.exe. Maybe we should include an aka.ms link to docs, where we can be wordy about it?

File: build/LegacyPackages/14.0/Microsoft.Build.Tasks.Core/Microsoft.Build.Tasks.Core.cs [hunk 2]
```diff
@@ -903,8 +903,11 @@ public sealed partial class ResolveCodeAnalysisRuleSet : Microsoft.Build.Tasks.T
         public string ResolvedCodeAnalysisRuleSet { get { throw null; } }
         public override bool Execute() { throw null; }
     }
-#if !NETSTANDARD
+#if NETSTANDARD
```
Comment (rainersigwald) [inline build/LegacyPackages/14.0/Microsoft.Build.Tasks.Core/Microsoft.Build.Tasks.Core.cs]: I thought the `LegacyPackages` folder was dead. Did something force you to do this, or were you being proactive? If the latter, probably just revert it -- we'll clean up later.

File: src/Tasks/Resources/Strings.resx (hunk 1 above)
Comment (nguerrera) [reply src/Tasks/Resources/Strings.resx]: cc @mairaw

Comment (mairaw) [reply src/Tasks/Resources/Strings.resx]: Yeah, that's tricky. I like the idea of linking to docs where this can be explained in more detail.

Comment (AaronRobinsonMSFT) [reply src/Tasks/Resources/Strings.resx]: How would I get that link?

File: build/LegacyPackages/14.0/Microsoft.Build.Tasks.Core/Microsoft.Build.Tasks.Core.cs (hunk 2 above)
Comment (AaronRobinsonMSFT) [reply build/LegacyPackages/14.0/Microsoft.Build.Tasks.Core/Microsoft.Build.Tasks.Core.cs]: Proactive 😄 Will revert and see what happens.

File: src/Tasks/Resources/Strings.resx (hunk 1 above)
Comment (mairaw) [reply src/Tasks/Resources/Strings.resx]: You can create one using http://aka.ms. You can put me as co-owner and ddfwlink as a security group.

Comment (AaronRobinsonMSFT) [reply src/Tasks/Resources/Strings.resx]: Yay. Okay, so now is there an existing doc I should link against?

Comment (mairaw) [reply src/Tasks/Resources/Strings.resx]: I don't think that explanation already exists. 
Maybe link to https://docs.microsoft.com/visualstudio/msbuild/msbuild for now.

I've filed a doc issue for the VS team: https://github.com/MicrosoftDocs/visualstudio-docs/issues/2016

File: src/Tasks/ResolveComReference.Stub.cs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    /// Main class for the COM reference resolution task
+    /// </summary>
+    public sealed partial class ResolveComReference : Microsoft.Build.Tasks.TaskExtension
+    {
+        #region Properties
+
+        /// <summary>
+        /// COM references specified by guid/version/lcid
+        /// </summary>
+        public ITaskItem[] TypeLibNames { get; set; }
```
[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply src/Tasks/ResolveComReference.Stub.cs]: Sure. I can create an interface to connect enforce this task's contract. Is there a place I should put this interface?

My plan would be to rename the files to be `ResolveComReference.framework.cs` and `ResolveComReference.core.cs`

File: src/Tasks/Resources/Strings.resx (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/Tasks/Resources/Strings.resx]: Thanks.

File: src/Tasks/ResolveComReference.Stub.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply src/Tasks/ResolveComReference.Stub.cs]: @rainersigwald I put everything in the existing `ResolveComReference.cs` file. Thoughts?

**Comment by rainersigwald** [comment]
Passed on rerun, evidently.  ¯\_(ツ)_/¯ I'll also poke CLA-bot.

**Comment by ccastanedaucf** [comment]
@AaronRobinsonMSFT ```ForceSomeOutOfDate``` seems to be a flaky test, it also randomly failed on me all the way back here https://github.com/Microsoft/msbuild/pull/3631#issuecomment-414465123 although I never ran into again

**Comment by AaronRobinsonMSFT** [comment]
@ccastanedaucf Thanks. It seems to be building now.

@rainersigwald or @nguerrera any other comments or concerns? I am good for this to be merged in.

File: src/Tasks/ResolveComReference.cs [hunk 4]
```diff
@@ -23,85 +21,213 @@
 
 namespace Microsoft.Build.Tasks
 {
-    /// <summary>
-    /// Main class for the COM reference resolution task
-    /// </summary>
-    public sealed partial class ResolveComReference : AppDomainIsolatedTaskExtension, IComReferenceResolver
+    internal interface IResolveComReferenceTaskContract
```
Comment (rainersigwald) [inline src/Tasks/ResolveComReference.cs]: I thought about rewriting to use 3 partial classes but this works 👍.

File: src/Tasks/Resources/Strings.resx [hunk 5]
```diff
@@ -2673,6 +2673,12 @@
     <comment>{StrBegin="MSB4801: "}</comment>
   </data>
 
+  <!-- Some tasks are only supported on .NET Framework -->
+  <data name="TaskRequiresFrameworkFailure" xml:space="preserve">
+    <value>MSB4803: The task "{0}" is not supported on the .NET Core version of MSBuild. Please use the .NET Framework version of MSBuild. See https://aka.ms/MSBuildProduct for further details.</value>
```
[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply src/Tasks/Resources/Strings.resx]: Um... There is no site for that error yet? This was a placeholder for MSBuild based on a suggestion from @mairaw. Once we get a doc for the error, we could update that?

[the remark this answers is not in the document]
Comment (rainersigwald) [reply src/Tasks/Resources/Strings.resx]: I think we can create a final(ish) URL and still link it to a placeholder destination. I (just) created https://aka.ms/msbuild/MSB4803.

[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply src/Tasks/Resources/Strings.resx]: Oh sorry. I misunderstood. Yep. That works. Thanks!

**Comment by rainersigwald** [comment]
Why, CLA-bot, why? (closing/reopening to poke it)

**Pull request merged**